### PR TITLE
*: fix update relay processing order

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 
in DM-worker `UpdateRelayConfig` function, the check of `update source ID is not allowed` is before new config loaded, which will always fail.
Besides I find we will get the following error after we update a relay config. This is caused by we forget to close binlog replication syncer after relay paused
```
2019/05/10 06:19:52.383 relay.go:176: [error] [relay] process exit with error ERROR 1236 (HY000): A slave with the same server_uuid/server_id as this slave has connected to the master; the first event 'bin.000007' at 27622680, the last event read from './bin.000007' at 27622993, the last byte read from './bin.000007' at 27622993.
github.com/pingcap/errors.AddStack
        /root/.gvm/pkgsets/go1.12/global/pkg/mod/github.com/pingcap/errors@v0.11.0/errors.go:174
github.com/pingcap/errors.Trace
        /root/.gvm/pkgsets/go1.12/global/pkg/mod/github.com/pingcap/errors@v0.11.0/juju_adaptor.go:12
github.com/pingcap/dm/relay.(*Relay).process
        /root/code/gopath/src/github.com/pingcap/dm/relay/relay.go:357
github.com/pingcap/dm/relay.(*Relay).Process
        /root/code/gopath/src/github.com/pingcap/dm/relay/relay.go:173
github.com/pingcap/dm/dm/worker.(*RelayHolder).run
        /root/code/gopath/src/github.com/pingcap/dm/dm/worker/relay.go:107
github.com/pingcap/dm/dm/worker.(*RelayHolder).resumeRelay.func1
        /root/code/gopath/src/github.com/pingcap/dm/dm/worker/relay.go:200
runtime.goexit
        /root/.gvm/gos/go1.12/src/runtime/asm_amd64.s:1337
2019/05/10 06:19:52.383 relay.go:113: [error] process error with type UnknownError:
```

### What is changed and how it works?

1. Load the new DM-worker config before any validation
2. close binlog syncer in `TCPReader.Close`

### Check List 

Tests 

 - Integration test
   - more integration test with dmctl API will be added in pr #135 
 - Manual test (add detailed scripts or steps below)
   - deploy a DM cluster
   - run `update-relay -w worker_host:worker_port path_to_new_dm_worker_config` in dmctl
   - check `update-relay` is success

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @GregoryIan @csuzhangxc

File: pkg/binlog/reader/tcp.go [hunk 1]
```diff
@@ -114,6 +114,16 @@ func (r *TCPReader) Close() error {
 		return errors.New("already closed")
 	}
 
+	setStageClosed := func() error {
+		r.stage = stageClosed
+		return nil
+	}
+
+	if r.syncer == nil {
```
Comment (csuzhangxc) [inline pkg/binlog/reader/tcp.go]: when `r.syncer == nil`?

Comment (amyangfei) [reply pkg/binlog/reader/tcp.go]: remove this check. If we use `NewTCPReader` to get a TCPReader, `syncer` is always created

**Comment by amyangfei** [comment]
`codecov diff hit` means the coverage rate of diff code, it is not stable now, may be we can ignore

**Pull request merged**